### PR TITLE
Fetch optical centre and focal length from gazebo directly. 

### PR DESCRIPTION
**Pull request by deepanshubansal01**
This PR gets the `optical centre` and `focal length` from gazebo directly instead of getting them from gazebo plugin tags specified by user. A deprecating warning is shown if the user any of the the `<cx>`, `<cy>` or `<focal_length>` tags. 

Depends on `gazebo` PR:  https://github.com/osrf/gazebo/pull/3245
Some related discussion here: https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1407

Signed-off-by: deepanshu [deepanshubansal01@gmail.com](mailto:deepanshubansal01@gmail.com)

### CODE REVIEW

**Comment by adityapande-1995** [comment]
CI will fail until we merge this one and make a new release of gazebo : https://github.com/osrf/gazebo/pull/3245

File: gazebo_plugins/src/gazebo_ros_camera.cpp [hunk 1]
```diff
@@ -356,32 +368,41 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
...
+    impl_->cx_.push_back(impl_->camera_[i]->ImageOpticalCenterX());
+    impl_->cy_.push_back(impl_->camera_[i]->ImageOpticalCenterY());
     impl_->hfov_.push_back(impl_->camera_[i]->HFOV().Radian());
 
-    double computed_focal_length =
-      (static_cast<double>(width[i])) / (2.0 * tan(impl_->hfov_[i] / 2.0));
+    if(_sdf->HasElement("cx")) {
+      RCLCPP_WARN(
+          impl_->ros_node_->get_logger(),
+          "The <cx> [%f] you have provided for camera [%s] is "
```
Comment (jacobperron) [inline gazebo_plugins/src/gazebo_ros_camera.cpp]:
```suggestion
          "The <cx> [%f] tag for camera [%s] is "
```

File: gazebo_plugins/src/gazebo_ros_camera.cpp [hunk 2]
```diff
@@ -356,32 +368,41 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
...
+          "deprecated and will be ignored. Consider not providing the "
```
Comment (jacobperron) [inline gazebo_plugins/src/gazebo_ros_camera.cpp]: I don't think we should ignore the tag. We should retain the exact same behavior as before, except with a deprecation message. The deprecation message should suggest what Gazebo tag to use alternatively.

**Review by jacobperron — CHANGES_REQUESTED** [review]

File: gazebo_plugins/src/gazebo_ros_camera.cpp (hunk 2 above)
Comment (deepanshubansal01) [reply gazebo_plugins/src/gazebo_ros_camera.cpp]: I see. I will make the changes. Thanks for the review.

**Review by jacobperron — APPROVED** [review]
LGTM pending CI

**Comment by scpeters** [comment]
preparing for gazebo 11.12.0 release in https://github.com/osrf/gazebo/pull/3260

**Comment by scpeters** [comment]
> preparing for gazebo 11.12.0 release in [osrf/gazebo#3260](https://github.com/osrf/gazebo/pull/3260)

gazebo 11.12.0 has been released, and I just resolved conflicts, so let's see what happens in CI

**Comment by scpeters** [comment]
> > preparing for gazebo 11.12.0 release in [osrf/gazebo#3260](https://github.com/osrf/gazebo/pull/3260)
> 
> gazebo 11.12.0 has been released, and I just resolved conflicts, so let's see what happens in CI

I forgot that we are no longer building new debs for 22.04 in packages.osrfoundation.org (see [this post](https://community.gazebosim.org/t/gazebo-classic-on-jammy/1248/2)), so the changes from https://github.com/osrf/gazebo/pull/3245 won't actually be usable. I'll think about what we should do. It's possible that we could use the feature from SDFormat in https://github.com/gazebosim/sdformat/pull/1088

**Comment by scpeters** [comment]
this isn't mergeable to `ros2` since the gazebo11 debs aren't being updated on 22.04

**Pull request closed without merging**